### PR TITLE
feature/load testing script

### LINKED ISSUE #154: LoadTest for Beta
Please write a script which creates ~500 clusters. Those clusters should be distributed across all available seed clusters.

If all clusters are up, create 3-5 nodes per cluster. For simplicity use just one provider.

All this should happen via script!

### PR DESCRIPTION
**Pull request by lbb**
Resolves #154

### CODE REVIEW

File: cmd/test-kubermatic-load/main.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kubermatic/api"
+)
+
+var (
+	jwtFlag         = flag.String("jwt", "", "The String of the Authorization: header")
+	maxNodesFlag    = flag.Int("nodes", 0, "Spcifies the amount of nodes to create in one cluster (nodes*clusters)")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: Typo

File: cmd/test-kubermatic-load/main.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+	maxClustersFlag = flag.Int("clusters", 0, "Spcifies the amount of clusters to deploy")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: Typo

File: cmd/test-kubermatic-load/main.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	dcFlag          = flag.String("datacenter", "master", "use this to specify a datacenter")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: Capitalization

File: cmd/test-kubermatic-load/main.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+	maxAsyncFlag    = flag.Int("max-async", 10, "Spcifies the amount of request running at the same time")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: Typo and Pluralization

File: cmd/test-kubermatic-load/main.go [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+		var clusterState api.Cluster
+		if err = json.NewDecoder(bytes.NewReader(data)).Decode(&clusterState); err != nil {
+			log.Println(string(data))
+			return err
+		}
+		if clusterState.Address.URL != "" && clusterState.Status.Phase == api.RunningClusterStatusPhase {
+			break
+		}
+		log.Println("Waiting for NS to get created ....")
+		time.Sleep(time.Second * 10)
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: Make this configurable

File: cmd/test-kubermatic-load/main.go [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+			<-done
+		}(cluster)
+	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	printError := func() {
+		log.Printf("Wrong usage usage.\n\n\t %s [up|purge]\n\n", os.Args[0])
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: Typo

**Review by scruplelesswizard — CHANGES_REQUESTED** [review]

File: cmd/test-kubermatic-load/main.go [hunk 7]
```diff
@@ -17,42 +17,48 @@ import (
 )
 
 var (
-	jwtFlag         = flag.String("jwt", "", "The String of the Authorization: header")
-	maxNodesFlag    = flag.Int("nodes", 0, "Spcifies the amount of nodes to create in one cluster (nodes*clusters)")
-	maxClustersFlag = flag.Int("clusters", 0, "Spcifies the amount of clusters to deploy")
-	dcFlag          = flag.String("datacenter", "master", "use this to specify a datacenter")
-	maxAsyncFlag    = flag.Int("max-async", 10, "Spcifies the amount of request running at the same time")
+	jwtFlag             = flag.String("jwt-auth", "", "The String of the authorization header")
+	maxNodesFlag        = flag.Int("node-count", 0, "The amount of nodes to create in one cluster (node-count*cluster-count)")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go:21]: `"The number of nodes to create per cluster"`

File: cmd/test-kubermatic-load/main.go [hunk 8]
```diff
@@ -17,42 +17,48 @@ import (
...
+	maxClustersFlag     = flag.Int("cluster-count", 0, "The amount of clusters to deploy")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go:22]: `"The number of clusters to create"`

File: cmd/test-kubermatic-load/main.go [hunk 9]
```diff
@@ -17,42 +17,48 @@ import (
...
+	dcFlag              = flag.String("datacenter-name", "master", "The master dc")
+	maxAsyncFlag        = flag.Int("max-workers", 10, "The amount of request running at the same time")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: `"Maximum concurrent requests"`

File: cmd/test-kubermatic-load/main.go [hunk 10]
```diff
@@ -17,42 +17,48 @@ import (
...
+	retryNSIntervalFlag = flag.Int64("ns-retry-interval", 10, "The amout of time until a NS alive request is send again")
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go]: `"Duration in seconds to wait between {I really have no idea what we are waiting between}"`

File: cmd/test-kubermatic-load/main.go [hunk 11]
```diff
@@ -17,42 +17,48 @@ import (
...
 )
 
+// setAuth sets the jwt token int a requests Authorization header field
```
Comment (scruplelesswizard) [inline cmd/test-kubermatic-load/main.go:29]: /int/in/

**Review by scruplelesswizard — CHANGES_REQUESTED** [review]
Please clarify the param descriptions as required

**Comment by scruplelesswizard** [comment]
We are still missing usage instructions. This will not be merged until we have some docs on how to use it.

**Review by scruplelesswizard — APPROVED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**